tests/modules/lib/cddl-gen/decode/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(NONE)
+
+find_program(CDDL_GEN cddl-gen)
+if (NOT CDDL_GEN)
+  message(FATAL_ERROR
+    'cddl-gen was not found. Install it via pip. This script expects cddl-gen version 0.3.0')
+endif()
+
+set(cddl_gen_args
+  -c ${CMAKE_CURRENT_LIST_DIR}/test.cddl
+  code
+  --output-c ${PROJECT_BINARY_DIR}/src/test_decode.c
+  --output-h ${PROJECT_BINARY_DIR}/include/test_decode.h
+  -t Test
+  -d
+  )
+
+execute_process(
+  COMMAND ${CDDL_GEN} ${cddl_gen_args}
+  COMMAND_ERROR_IS_FATAL ANY
+  )
```

Comment:
Why are you using `execute_process()` here instead of `add_custom_command()`.

`add_custom_command()` with proper OUTPUT and DEPENDS allows CMake to setup correctly dependencies and GENERATED property, which ensures correct behavior, as well as files are cleaned correctly on a `nimja/make clean`.

https://cmake.org/cmake/help/latest/prop_sf/GENERATED.html

Reply:
Thanks, changed this.